### PR TITLE
Add windows servers to smoke test

### PR DESCRIPTION
**Pull request by yosifkit**
Disable cgo to fix build failure on GHA `windows-2022`

Example failure from PRs to official-images:
```console
go: downloading github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da
# github.com/docker-library/bashbrew/cmd/bashbrew
C:\hostedtoolcache\windows\go\1.18.10\x64\pkg\tool\windows_amd64\link.exe: running gcc failed: exit status 1
C:/ProgramData/Chocolatey/lib/mingw/tools/install/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Users\RUNNER~1\AppData\Local\Temp\go-link-3224421645\000008.o: in function `_cgo_preinit_init':
\\_\_\runtime\cgo/gcc_libinit_windows.c:30: undefined reference to `__imp___iob_func'
C:/ProgramData/Chocolatey/lib/mingw/tools/install/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Users\RUNNER~1\AppData\Local\Temp\go-link-3224421645\000008.o: in function `x_cgo_sys_thread_create':
\\_\_\runtime\cgo/gcc_libinit_windows.c:60: undefined reference to `__imp___iob_func'
C:/ProgramData/Chocolatey/lib/mingw/tools/install/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Users\RUNNER~1\AppData\Local\Temp\go-link-3224421645\000008.o: in function `x_cgo_notify_runtime_init_done':
\\_\_\runtime\cgo/gcc_libinit_windows.c:101: undefined reference to `__imp___iob_func'
C:/ProgramData/Chocolatey/lib/mingw/tools/install/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Users\RUNNER~1\AppData\Local\Temp\go-link-3224421645\000009.o: in function `x_cgo_thread_start':
\\_\_\runtime\cgo/gcc_util.c:18: undefined reference to `__imp___iob_func'
C:/ProgramData/Chocolatey/lib/mingw/tools/install/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Users\RUNNER~1\AppData\Local\Temp\go-link-3224421645\000010.o: in function `_cgo_sys_thread_start':
\\_\_\runtime\cgo/gcc_windows_amd64.c:31: undefined reference to `__imp___iob_func'
collect2.exe: error: ld returned 1 exit status
```

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -11,6 +11,8 @@ runs:
     - uses: actions/setup-go@v3
       with:
         go-version-file: '${{ github.action_path }}/go.mod'
+      env:
+        CGO_ENABLED: 0
```
Comment (tianon) [inline action.yml:15]: Should we instead add this to https://github.com/docker-library/bashbrew/blob/6334a4f8e8b210d771d5ba55755f321e57a4ca08/bashbrew.sh#L9 ?

(with `: "${CGO_ENABLED:=0}"`)

Comment (yosifkit) [reply action.yml:15]: Yeah I think that makes sense.
```console
$ git grep -i cgo
Dockerfile:RUN CGO_ENABLED=0 ./bashbrew.sh --version; \
Dockerfile.release:ENV CGO_ENABLED 0
action.yml:        CGO_ENABLED: 0
```

Comment (yosifkit) [reply action.yml]: Ok, now moved to `bashbrew.sh`. We could drop it from the `Dockerfile` but it seems fine to be explicit there and in the `Dockerfile.release`.
```console
$ git grep -i cgo
Dockerfile:RUN CGO_ENABLED=0 ./bashbrew.sh --version; \
Dockerfile.release:ENV CGO_ENABLED 0
bashbrew.sh:: "${CGO_ENABLED:=0}"
bashbrew.sh:export GO111MODULE=on CGO_ENABLED
```

**Review by tianon — APPROVED** [review]

**Comment by whalelines** [comment]
Is there a reason to disable for all rather than just for `windows`?

**Comment by yosifkit** [comment]
Just for consistency across OS, architecture, and build pipeline. It was already disabled for the builds using `Dockerfile` or `Dockerfile.release`, just not the script directly on a host like GitHub Actions.

**Pull request merged**